src/components/Swap/SwapButton/index.tsx

```diff
@@ -118,11 +122,14 @@ export default memo(function SwapButton({ disabled }: SwapButtonProps) {
       })
       setDisplayTxHash(transaction.hash)
 
-      // Set the block containing the response to the oldest valid block to ensure that the
-      // completed trade's impact is reflected in future fetched trades.
-      transaction.wait(1).then((receipt) => {
-        setOldestValidBlock(receipt.blockNumber)
-      })
+      transaction
+        .wait(1)
+        .then((receipt) => {
+          // Set the block containing the response to the oldest valid block to ensure that the
+          // completed trade's impact is reflected in future fetched trades.
+          setOldestValidBlock(receipt.blockNumber)
+        })
+        .catch((e) => onTxFail?.(e, transaction.hash)) // Do we run onTxFail on tx cancelled?
```

Comment:
this isn't necessarily an ethereum tx fail — the ethers library throws an error on cancelled tx, which is what we're catching here

Reply:
I don't think we should call `onTxFail` in this case. @akarys92 thoughts?

(aside-in the cancellation case we wouldn't have a `transaction.hash` here)